contrib/generic_proxy/filters/network/source/route.cc

```diff
@@ -88,6 +87,114 @@ RouteEntryConstSharedPtr RouteMatcherImpl::routeEntry(const Request& request) co
   return nullptr;
 }
 
+RouteMatcherImpl::RouteMatcherImpl(const ProtoRouteConfiguration& route_config,
+                                   Envoy::Server::Configuration::ServerFactoryContext& context,
+                                   bool validate_clusters)
+    : name_(route_config.name()) {
+  for (const auto& virtual_host_config : route_config.virtual_hosts()) {
```

Comment:
Is there any way to share the code with HCM routes? I checked the code is almost same

Reply:
Hmmm, it's a little hard. May be some templates could be used here. If you think it's ok I will add a TODO here. Because I don't want change too much code of core code base in thie contrib update.

Reply:
agree with that not change the core code.